tests/test_nblist.py

```diff
@@ -19,47 +20,84 @@ def test_empty_neighborlist():
         custom_ops.Neighborlist_f32(0)
 
 
-def test_block_bounds():
-    np.random.seed(2020)
-    sizes = [128, 156, 298]
-    max_size = max(sizes)
-    nblist = custom_ops.Neighborlist_f64(max_size)
-    for N in sizes:
-        nblist.resize(N)
-        block_size = 32
+def reference_block_bounds(coords: NDArray, box: NDArray, block_size: int) -> Tuple[NDArray, NDArray]:
+    # Make a copy to avoid modify the coordinates that end up used later by the Neighborlist
+    coords = coords.copy()
```

Comment:
If you don't have this copy, you can end up re-centering all of the coordinates which results in not testing the PBCs in the kernel. This is an issue on master, but thankfully the kernel was implemented correctly. But when the kernel incorrectly handled PBCs, tests would still pass. With this copy and a bad implementation, it does correctly fail.

Reply:
Great catch!